src/hooks/BaseTransferHook.sol

```diff
@@ -40,6 +42,8 @@ abstract contract BaseTransferHook is Auth, IMemberlist, IFreezable, ITransferHo
         address crosschainSource_,
         address deployer
     ) Auth(deployer) {
+        require(redeemSource_ != depositTarget_ && depositTarget_ != crosschainSource_, InvalidInputs());
```

Comment:
Do we also want to restrict redeemSource != crosschainSource?

Reply:
Yeah we should true